features/environment.py

```diff
@@ -412,6 +412,59 @@ def before_scenario(context: Context, scenario: Scenario):
             )
 
 
+def after_scenario(context, scenario):
+    """Collect the coverage files after the scenario is run."""
+    cov_dir = os.path.join(context.pro_config.artifact_dir, "coverage")
+    if not os.path.exists(cov_dir):
+        os.makedirs(cov_dir)
+
+    inner_dir = os.path.join(
+        datetime.datetime.now().strftime("%Y-%m-%d"),
+        "{}".format(os.path.basename(scenario.filename.replace(".", "_"))),
+    )
+    new_artifacts_dir = os.path.join(
+        cov_dir,
+        inner_dir,
+    )
+    if not os.path.exists(new_artifacts_dir):
+        os.makedirs(new_artifacts_dir)
+    if hasattr(context, "machines") and SUT in context.machines:
+        try:
+            scenario_name = (
+                os.path.basename(scenario.filename.replace(".", "_"))
+                + "_"
+                + str(scenario.line)
+            )
+            cov_filename = ".coverage.{}".format(scenario_name)
+            context.machines[SUT].instance.execute(
+                [
+                    "bash",
+                    "-c",
+                    "mv .coverage /tmp/{cov_filename}".format(
+                        cov_filename=cov_filename
+                    ),
+                ],
+                use_sudo=True,
+            )
+            context.machines[SUT].instance.execute(
+                [
+                    "chmod",
+                    "666",
+                    "/tmp/{cov_filename}".format(cov_filename=cov_filename),
+                ],
+                use_sudo=True,
+            )
+
+            dest = os.path.join(new_artifacts_dir, cov_filename)
+            context.machines[SUT].instance.pull_file(
+                "/tmp/{cov_filename}".format(cov_filename=cov_filename), dest
```

Comment:
We construct this string three times in a row. Consider putting it in a variable and re-using it